File: demo-test/src/test/java/io/fries/demo/test/cucumber/world/WorldClock.java
package io.fries.demo.test.cucumber.world;

import java.time.ZonedDateTime;
import java.util.function.Supplier;

public class WorldClock {

    private Supplier<ZonedDateTime> clock;

    private WorldClock() {
        clock = ZonedDateTime::now;
    }

    public static WorldClock reset() {
        return new WorldClock();
    }

    public void fixedAt(final ZonedDateTime zonedDateTime) {
        clock = () -> zonedDateTime;
    }

    public ZonedDateTime now() {
        return clock.get();
    }
}

File: demo-test/src/test/kotlin/io/fries/demo/test/cucumber/world/WorldClock.kt
package io.fries.demo.test.cucumber.world

import java.time.ZonedDateTime

class WorldClock(
    private var clock: () -> ZonedDateTime = { ZonedDateTime.now() }
) {

    fun fixedAt(zonedDateTime: ZonedDateTime) = apply {
        clock = { zonedDateTime }
    }

    fun now(): ZonedDateTime = clock.invoke()
}